kotlintest-core/src/main/kotlin/io/kotlintest/specs/AbstractAnnotationSpec.kt

```diff
@@ -13,8 +30,247 @@ abstract class AbstractAnnotationSpec(body: AbstractAnnotationSpec.() -> Unit =
+          "This annotation is here to provide easy migration from JUnit to KotlinTest. Use BeforeTest instead.",
+          ReplaceWith("BeforeTest")
+  )
+  annotation class Before
+
+
+  // Before Spec annotations
+
+  /**
+   * Marks a function to be executed before each spec
+   *
+   * This can be used in AnnotationSpec to mark a function to be executed before a spec by KotlinTest Engine.
+   * @see BeforeTest
+   */
+  annotation class BeforeSpec
+
+  /**
+   * Identical to [BeforeSpec]
+   */
+  @Deprecated(
+          "This annotation is here to provide easy migration from JUnit to Kotlintest. Use BeforeSpec instead.",
+          ReplaceWith("BeforeSpec")
+  )
+  annotation class BeforeAll
+
+  /**
+   * Identical to [BeforeSpec]
+   */
+  @Deprecated(
+          "This annotation is here to provide easy migration from JUnit to Kotlintest. Use BeforeSpec instead.",
+          ReplaceWith("BeforeSpec")
+  )
+  annotation class BeforeClass
+
+
+  // After Test annotations
+
+  /**
+   * Marks a function to be executed after each test
+   *
+   * This can be used in AnnotationSpec to mark a function to be executed before a test by KotlinTest Engine.
+   * @see afterSpec
+   */
+  annotation class AfterTest
+
+  /**
+   * Identical to [AfterTest]
+   */
+  @Deprecated(
+          "This annotation is here to provide easy migration from JUnit to KotlinTest. Use AfterTest instead.",
+          ReplaceWith("AfterTest")
+  )
+  annotation class AfterEach
+
+  /**
+   * Identical to [AfterTest]
+   */
+  @Deprecated(
+          "This annotation is here to provide easy migration from JUnit to KotlinTest. Use AfterTest instead.",
+          ReplaceWith("AfterTest")
+  )
+  annotation class After
+
+
+  // After Spec annotations
+
+  /**
+   * Marks a function to be executed after each spec
+   *
+   * This can be used in AnnotationSpec to mark a function to be executed before a spec by KotlinTest Engine.
+   * @see [AfterTest]
+   */
+  annotation class AfterSpec
+
+  /**
+   * Identical to [AfterSpec]
+   */
+  @Deprecated(
+          "This annotation is here to provide easy migration from JUnit to KotlinTest. Use AfterSpec instead.",
+          ReplaceWith("AfterSpec")
+  )
+  annotation class AfterAll
+
+  /**
+   * Identical to [AfterSpec]
+   */
+  @Deprecated(
+          "This annotation is here to provide easy migration from JUnit to KotlinTest. Use AfterSpec instead.",
+          ReplaceWith("AfterSpec")
+  )
+  annotation class AfterClass
+
+  // Test specific annotations
+  /**
+   * Marks a function to be executed as a Test
+   *
+   * This can be used in AnnotationSpec to mark a function to be executed as a test by KotlinTest Engine.
+   */
+  annotation class Test
+
+  /**
+   * Marks a Test to be ignored
+   *
+   * This can be used in AnnotationSpec to mark a Test as Ignored.
+   */
+  annotation class Ignore
+
+}
+
+@Suppress("DEPRECATION")
+fun KClass<out AbstractAnnotationSpec>.findBeforeTestFunctions(): List<KFunction<*>> {
```

Comment:
I thought of marking these functions as internal, but they wouldn't be available in `SpecRunner`